Add How-To Guides (#15)
pace-neutrons/resolution_functions: docs/source/howtos/programmatic_query.rst

the default version for that instrument. Though, if the instrument being
searched for is an alias for a particular version of an instrument, that version
is returned instead of the default version:

>>> available_versions, default_version = Instrument.available_versions('TFXA')
>>> available_versions
['TFXA', 'TOSCA1', 'TOSCA']
>>> default_version
'TFXA'

.. _how-to-model:

How To Find the Models Available for an Instrument
--------------------------------------------------

Given an :ref:`instrument<how-to-instrument>` and its
:ref:`version<how-to-version>`, it is possible to query the list of available
:term:`models<model>` by using the
:py:meth:`~resolution_functions.instrument.Instrument.available_models`
property:

>>> from resolution_functions import Instrument
>>> tosca = Instrument.from_default('TOSCA', 'TOSCA')
>>> tosca.available_models
['AbINS', 'book', 'vision']

The default model can similarly be accessed via an attribute:

>>> tosca.default_model
'AbINS'

Advanced
^^^^^^^^

The above property returns only the recommended (usually latest) version of each
model. These "models" are actually aliases that each point to a versioned model

> that actually holds the data. These versioned models can be listed by using

Reviewer: ```suggestion
which holds the data. These versioned models can be listed by using
```

:py:meth:`~resolution_functions.instrument.Instrument.available_unique_models`:

>>> tosca.available_unique_models
['AbINS_v1', 'book_v1', 'vision_v1']

Then, to round out the options, *all* models can be listed via
:py:meth:`~resolution_functions.instrument.Instrument.all_available_models`:

>>> tosca.all_available_models
['AbINS', 'AbINS_v1', 'book', 'book_v1', 'vision', 'vision_v1']

.. important::

>     Older versions of models often contain bugs and/or inaccuracies - use at own

Reviewer: ```suggestion
    Older versions of models may contain known bugs and/or inaccuracies - use at own
```

    risk.

.. _how-to-config:

How To Find the Configurations that Must be Chosen for a Model
--------------------------------------------------------------

To do this, the :term:`model` must already be known, either from prior knowledge
or by browsing the :ref:`available models<how-to-model>`; without this decision

> made, the only option is use :ref:`howto-query-models-configs` which returns a

Reviewer: "The only option is..." followed by "Otherwise you can do X"? Seems wrong.

Author: "without this decision made, the only option is use" -> then "otherwise" implies the other case, i.e. when the decision (about which model to use) *is* made. Maybe could use some rewording to make it clearer? E.g. "Otherwise" -> "In the former case - with the model known - ..."

Reviewer: Maybe the whole paragraph can just be

> The :term:`configuration<configuration>` argument(s) required by a given :term:`model` can be retrieved using the :py:meth:`~resolution_functions.instrument.Instrument.possible_configurations_for_model` method.

The section headings makes the requirements pretty clear, really.

Author: I like to create a cross-link with the related section (the one that returns a table of models and configurations), tho. Do you think that's superfluous? Maybe I could do a `.. seealso::` section below the example...

Reviewer: It's helpful where it fits the flow, but here it is just a distraction that makes the explanation more confusing.

table. Otherwise, the :term:`configurations<configuration>` can be retrieved
using the
:py:meth:`~resolution_functions.instrument.Instrument.possible_configurations_for_model`
method:

>>> from resolution_functions import Instrument
>>> tosca = Instrument.from_default('TOSCA', 'TOSCA')
>>> tosca.possible_configurations_for_model('AbINS')
[]
>>> tosca.possible_configurations_for_model('book')
['detector_bank']

.. _howto-query-options:

How To Find the Options Available for a Configuration
-----------------------------------------------------

> Similarly, to obtain the :term:`options<option>` in a Python format, the
> :term:`model` must once again be known
> (:ref:`otherwise<howto-query-models-configs-options>` only a formatted table can
> be returned), though in this case there are multiple options:

Reviewer: ```suggestion
To obtain the :term:`options<option>` as Python data, the
:term:`model` must already be known.
(Humans can read the formatted table from :ref:`otherwise<howto-query-models-configs-options>`.)
```
